roles/katello_client/tasks/main.yml

```
@@ -29,7 +29,7 @@
     server_insecure: "{{ katello_client_insecure }}"
   ignore_errors: "{{ katello_client_ignore_registration_errors }}"
   when:
-    - katello_client_activationkey != ""
+    - katello_client_activationkey
```

```suggestion
    - katello_client_activationkey is defined
```

otherwise it will error in the case it's not defined as undef doesn't cast to bool